broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageStateController.java

```diff
@@ -294,7 +340,13 @@ public void updateCommandSentTime(final MessageSubscription subscription) {
 
   public List<MessageSubscription> findSubscriptions(
       final DirectBuffer messageName, final DirectBuffer correlationKey) {
-    return subscriptionState.findSubscriptions(messageName, correlationKey);
+    // ignore subscriptions for which a correlate command is already send for (i.e. pending
+    // subscription) - will be improved by #1421
+    return subscriptionState
+        .findSubscriptions(messageName, correlationKey)
+        .stream()
```

Comment:
Can we avoid to create garbage?

Reply:
Agree. I'll clean it up with #1421. Promise!